deepchecks/utils/distribution/drift.py

```diff
@@ -92,34 +111,54 @@ def cramers_v(dist1: Union[np.ndarray, pd.Series], dist2: Union[np.ndarray, pd.S
     phi2 = chi2 / n
     r, k = contingency_matrix.shape
     phi2corr = max(0, phi2 - ((k - 1) * (r - 1)) / (n - 1))
-    rcorr = r - ((r - 1)**2) / (n - 1)
-    kcorr = k - ((k - 1)**2) / (n - 1)
+    rcorr = r - ((r - 1) ** 2) / (n - 1)
+    kcorr = k - ((k - 1) ** 2) / (n - 1)
     return np.sqrt(phi2corr / min((kcorr - 1), (rcorr - 1)))
 
 
-def psi(expected_percents: np.ndarray, actual_percents: np.ndarray):
+def psi(dist1: Union[np.ndarray, pd.Series], dist2: Union[np.ndarray, pd.Series],
+        min_category_size_ratio: float = 0.01, max_num_categories: int = None, sort_by: str = 'difference') -> float:
     """
     Calculate the PSI (Population Stability Index).
 
     See https://www.lexjansen.com/wuss/2017/47_Final_Paper_PDF.pdf
 
     Parameters
     ----------
-    expected_percents: np.ndarray
-        array of percentages of each value in the expected distribution.
-    actual_percents: : np.ndarray
-        array of percentages of each value in the actual distribution.
+    dist1 : Union[np.ndarray, pd.Series]
+        array of numerical values.
+    dist2 : Union[np.ndarray, pd.Series]
+        array of numerical values to compare dist1 to.
+    min_category_size_ratio: float, default 0.01
+        minimum size ration for categories. Categories size ratio than this number are binned into an "Other" category.
+    max_num_categories: int, default: None
+        max number of allowed categories. If there are more categories than this number, categories are ordered by
+        magnitude and all the smaller categories are binned into an "Other" category.
+        If max_num_categories=None, there is no limit.
+        > Note that if this parameter is used, the ordering of categories (and by extension, the decision which
+        categories are kept by name and which are binned to the "Other" category) is done by default according to the
+        values of dist1, which is treated as the "expected" distribution. This behavior can be changed by using the
+        sort_by parameter.
+    sort_by: str, default: 'difference'
+        Specify how categories should be sorted, affecting which categories will get into the "Other" category.
+        Possible values:
+        - 'dist1': Sort by the largest dist1 categories.
+        - 'dist2': Sort by the largest dist2 categories.
+        - 'difference': Sort by the largest difference between categories.
+        > Note that this parameter has no effect if max_num_categories = None or there are not enough unique categories.
     Returns
     -------
     psi
         The PSI score
-
     """
+    expected_counts, actual_counts, _ = preprocess_2_cat_cols_to_same_bins(dist1, dist2, min_category_size_ratio,
+                                                                           max_num_categories, sort_by)
```

Comment:
I see here you inserted the preprocessing inside. Then again - as the preprocessing for PSI and Cramer's V is identical, I suggest you put it all outside.

Reply:
Why? Because there's a lot of code and doc duplication

Reply:
Change defaults to None